app/assets/javascripts/admin/user/experience_modal_view.js

```diff
@@ -2,152 +2,433 @@
-    this.setExperience(null, true);
-  };
-
-  setExperience = (event: ?SyntheticInputEvent<>, shouldAddValue: boolean = false): void => {
-    const { domain, level } = this.state;
-    if (domain && level !== null) {
-      const newUserPromises = this.props.users.map(user => {
-        if (this.props.selectedUserIds.includes(user.id)) {
-          let newExperienceLevel = parseInt(level);
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      multipleUsersEntries: this.loadMultipleUserEntries(props.selectedUsers),
+      singleUsersEntries: this.loadSingleUserEntries(props.selectedUsers),
+      showOnlySharedExperiences: false,
+    };
+  }
 
-          if (shouldAddValue && user.experiences[domain]) {
-            newExperienceLevel = user.experiences[domain] + parseInt(level);
-          }
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.visible && !this.props.visible) {
+      this.setState({
+        multipleUsersEntries: this.loadMultipleUserEntries(nextProps.selectedUsers),
+        singleUsersEntries: this.loadSingleUserEntries(nextProps.selectedUsers),
+      });
+    }
+  }
 
-          const newUser = update(user, {
-            experiences: { [domain]: { $set: newExperienceLevel } },
+  loadMultipleUserEntries = (users: Array<APIUserType>): Array<MultipleUserTableEntry> => {
+    if (users.length <= 1) {
+      return [];
+    }
+    // find all existing experience domains
+    let sharedTableEntries: Array<MultipleUserTableEntry> = [];
+    users.forEach(user => {
+      Object.keys(user.experiences).forEach(experience => {
+        if (!sharedTableEntries.find(entry => entry.domain === experience)) {
+          sharedTableEntries.push({
+            domain: experience,
+            value: 0,
+            lowestValue: 0,
+            highestValue: -1,
+            isShared: true,
           });
-
-          return this.sendUserToServer(newUser, user);
         }
-        return Promise.resolve(user);
       });
-
-      this.closeModal(newUserPromises);
-    }
+    });
+    // create sharedTableEntries
+    sharedTableEntries = sharedTableEntries.map(entry => {
+      let highestValue = -1;
+      let lowestValue = -1;
+      let isShared = true;
+      const domain = entry.domain;
+      users.forEach(user => {
+        const experiences = user.experiences;
+        if (domain in experiences) {
+          highestValue = highestValue > experiences[domain] ? highestValue : experiences[domain];
+          lowestValue =
+            lowestValue < experiences[domain] && lowestValue !== -1
+              ? lowestValue
+              : experiences[domain];
+        } else {
+          isShared = false;
+          lowestValue = 0;
+        }
+      });
+      return { domain, highestValue, lowestValue, value: lowestValue, isShared };
+    });
+    // sort entries
+    return sharedTableEntries.sort(
+      Utils.localeCompareBy(([]: Array<MultipleUserTableEntry>), entry =>
```

Comment:
This sort code appears four times in this module. Can you extract this into a helper method so that you only need to write `sort(sharedTableEntries)`?